doc/cookbook/pagination/Pagination.lhs

```diff
@@ -0,0 +1,251 @@
+
+#### Server
+
+Time to connect the last bits by defining the server implementation of our colorful API. The `Ranges`
+type we've defined above (tight to the `Range` HTTP header) indicates the server to parse any `Range`
+header, looking for the format defined in introduction with fields and target types we have just declared.
+If no such header is provided, we will end up receiving `Nothing`. Otherwise, it will be possible
+to _extract_ a `Range` from our `Ranges`.
+
+``` haskell
+server :: Server API
+server = handler
+  where
+    handler :: Maybe (Ranges '["name"] Color) -> Handler (Headers MyHeaders [Color])
+    handler mrange = do
+      let range =
+            fromMaybe defaultRange (mrange >>= extractRange)
+
+      addHeader (length colors) <$> returnRange range (applyRange range colors)
+
+main :: IO ()
+main =
+  Warp.run 1442 $ Servant.serve (Proxy @API) server
+```
+
+Let's try it out using different ranges to observe the server's behavior. As a reminder, here's
+the format we defined, where `<field>` here can only be `name` and `<value>` must parse to a `String`:
+
+- `Range: <field> [<value>][; offset <o>][; limit <l>][; order <asc|desc>]`
+
+Beside the target field, everything is pretty much optional in the `Range` HTTP header. Missing parts
+are deducted from the `RangeOptions` that are part of the `HasPagination` instance. Therefore, all
+following examples are valid requests to send to our server:
+
+- 1 - `curl http://localhost:1442/colors -vH 'Range: name'`
+- 2 - `curl http://localhost:1442/colors -vH 'Range: name; limit 2'`
+- 3 - `curl http://localhost:1442/colors -vH 'Range: name Green; order asc; offset 1'`
+
+Considering the following default options:
+
+- `defaultRangeLimit: 100`
+- `defaultRangeOffset: 0`
+- `defaultRangeOrder: RangeDesc`
+
+The previous ranges reads as follows:
+
+- 1 - The first 100 colors, ordered by descending names
+- 2 - The first 2 colors, ordered by descending names
+- 3 - The 100 colors after `Green` (not included), ordered by ascending names.
+
+
+## Going Forward
+
+####  Multiple Ranges
+
+Note that in the simple above scenario, there's no ambiguity with `extractRange` and `returnRange`
+because there's only one possible `Range` defined on our resource. Yet, as you've most probably
+noticed, the `Ranges` combinator accepts a list of fields, each of which must declare a `HasPagination`
+instance. Doing so will make the other helper functions more ambiguous and type annotation are
+highly likely to be needed.
+
+
+``` haskell
+instance HasPagination Color "hex" where
+  type RangeType Color "hex" = String
+  getFieldValue _ = hex
+
+-- to then define: Ranges '["name", "hex"] Color
+```
+
+
+#### Parsing Options
+
+By default, `servant-pagination` provides an implementation of `getRangeOptions` for each
+`HasPagination` type-class. However, this can be overwritten when defining a instance of that
```

Comment:
for each `HasPagination` **instance**, I suppose?

Reply:
Indeed!

*hide in shame*

Reply:
Fixed in 93838ae9e76ffa18dcf4ff344b52decccd5a6db2